sklearn/utils/tests/test_validation.py

```diff
@@ -52,6 +52,18 @@ def test_as_float_array():
     assert_true(as_float_array(X, False) is not X)
     # Checking that the new type is ok
     np.testing.assert_equal(X2.dtype, np.float64)
+    # Checking that smaller ints are properly converted to float32
+    for intsize in (np.int8,
```

Comment:
Define a list of types, more readable this way I feel:

```py
tested_dtypes = [np.int8, np.int16, np.int32, np.uint8, np.uint16, np.uint32]
for dt in tested_dtypes:
    ...
```
If you could add np.bool to the tested_dtypes, it would be great.